### PR TITLE
Footer styling

### LINKED ISSUE #7: Footer Styling
Style the footer according to original website, layout, positioning, links, etc. Don't forget mobile.

### PR DESCRIPTION
**Pull request by cuzuco2**
Fix #7

Test URLs:
- Before: https://main--elco-aerotop-sg--hlxsites.hlx.page/de/
- After: https://footer-styling--elco-aerotop-sg--hlxsites.hlx.page/de/

### CODE REVIEW

**Comment by cuzuco2** [comment]
@ekremney @rofe 
Please review

File: blocks/footer/footer.css [hunk 1]
```diff
@@ -1,18 +1,80 @@
 footer {
-  padding: 2rem;
-  background-color: var(--overlay-background-color);
+  padding: 2rem;   
+  background-color: #f0f0f0;
```
Comment (solaris007) [inline blocks/footer/footer.css]: Please use the `--overlay-background-color` from the base styles.

File: blocks/footer/footer.css [hunk 2]
```diff
@@ -1,18 +1,80 @@
 footer {
-  padding: 2rem;
-  background-color: var(--overlay-background-color);
+  padding: 2rem;
```
Comment (solaris007) [inline blocks/footer/footer.css]: trailing spaces?

File: blocks/footer/footer.css [hunk 3]
```diff
@@ -1,18 +1,80 @@
...
-  background-color: var(--overlay-background-color);
+  padding: 2rem;   
+  background-color: #f0f0f0;
   font-size: var(--body-font-size-s);
 }
 
 footer .footer {
   max-width: 1200px;
-  margin: auto;
+  border:0; margin:0 auto; padding:0;
```
Comment (solaris007) [inline blocks/footer/footer.css]: while we don't have those rules in the linting config yet, please:
- one line per property
- space after colon

File: blocks/footer/footer.css [hunk 4]
```diff
@@ -1,18 +1,80 @@
...
+}
+
+.footer > div > div {
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+  justify-content:center;
+}
+
+footer .footer ul {
```
Comment (solaris007) [inline blocks/footer/footer.css]: for all classes, you can probably remove `footer` as `.footer`(dot) is specific enough

File: blocks/footer/footer.css [hunk 5]
```diff
@@ -1,18 +1,80 @@
...
+  display: flex;
+  overflow: hidden;
+  flex-flow: row wrap;
+  justify-content: flex-end;
+  flex-grow: 1;
+}
+
+footer .footer li {
+  border:0; margin:0; 
+  font-size: 14px;
```
Comment (solaris007) [inline blocks/footer/footer.css]: Please use `--body-font-size-s` or similar for font sizes

File: blocks/footer/footer.css [hunk 6]
```diff
@@ -1,18 +1,80 @@
...
+  vertical-align: middle;
+  font-size: 14px;
+  text-align: center;
+  flex-grow: 1;
 }
 
-footer .footer a {
-  text-transform: uppercase;
+@media only screen and (max-width: 781px) {
+  footer .footer p {padding-left: 0;}
```
Comment (solaris007) [inline blocks/footer/footer.css]: please use spaces after and before start/closing curly braces

File: blocks/footer/footer.js [hunk 7]
```diff
@@ -16,4 +16,11 @@ export default async function decorate(block) {
   footer.innerHTML = html;
   await decorateIcons(footer);
   block.append(footer);
+  const links = document.querySelectorAll('.footer a');
```
Comment (solaris007) [inline blocks/footer/footer.js]: nitpicking: you could encapsulate the logic in a dedicated function and call it here... so we keep the decorate method clean.

File: blocks/footer/footer.css [hunk 8]
```diff
@@ -1,18 +1,80 @@
...
+  line-height: 2rem;
+  vertical-align: middle;
+  font-size: 14px;
+  text-align: center;
+  flex-grow: 1;
 }
 
-footer .footer a {
-  text-transform: uppercase;
+@media only screen and (max-width: 781px) {
```
Comment (solaris007) [inline blocks/footer/footer.css]: where's the 781px coming from?

File: blocks/footer/footer.css [hunk 9]
```diff
@@ -1,18 +1,80 @@
...
+}
+
+footer .footer a:focus {
+  color: #bd3826;
+  outline: none;
 }
 
 footer .footer p {
   margin: 0;
+  padding-left: 10%;
```
Comment (solaris007) [inline blocks/footer/footer.css]: it seems padding is different on the original. any reason for using relative and the same all around?

Screenshot 2023-01-16 at 15 27 12
(left your's, right original)

File: blocks/footer/footer.css [hunk 10]
```diff
@@ -1,18 +1,80 @@
...
+  line-height: 2rem;
+  vertical-align: middle;
+  font-size: 14px;
```
Comment (solaris007) [inline blocks/footer/footer.css]: Same as before, use variables from base styles

File: blocks/footer/footer.css [hunk 11]
```diff
@@ -1,18 +1,80 @@
...
+
+footer .footer  a {
+  font-size: 14px;
+  text-decoration: none;
+  color: black;
+  text-transform: uppercase;    
+}
+   
+footer .footer a:hover {
+  color: #bd3826;
```
Comment (solaris007) [inline blocks/footer/footer.css]: this color is available in the base styles...

File: blocks/footer/footer.css [hunk 12]
```diff
@@ -1,18 +1,80 @@
...
+}
+
+footer .footer a.active {
+  color: #bd3826;
```
Comment (solaris007) [inline blocks/footer/footer.css]: this color is available in the base styles...

File: blocks/footer/footer.css [hunk 13]
```diff
@@ -1,18 +1,80 @@
...
+}
+
+footer .footer a:focus {
+  color: #bd3826;
```
Comment (solaris007) [inline blocks/footer/footer.css]: this color is available in the base styles...

File: blocks/footer/footer.css [hunk 14]
```diff
@@ -1,18 +1,80 @@
...
+  text-align: center;
+  border-right: 1px solid black;
+}
+
+footer .footer li:last-child {
+  border: 0; 
+}
+
+footer .footer  a {
+  font-size: 14px;
```
Comment (solaris007) [inline blocks/footer/footer.css]: Same as before, use variables from base styles

File: blocks/footer/footer.css [hunk 15]
```diff
@@ -1,18 +1,80 @@
...
+  text-decoration: none;
+  color: black;
```
Comment (solaris007) [inline blocks/footer/footer.css]: this color is available in the base styles...

File: blocks/footer/footer.css (hunk 1 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Done

File: blocks/footer/footer.css (hunk 2 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Padding is needed to have the needed height of the footer

File: blocks/footer/footer.css (hunk 3 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Done

File: blocks/footer/footer.css (hunk 4 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Ok, done

File: blocks/footer/footer.css (hunk 1 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Yes, done

File: blocks/footer/footer.css (hunk 5 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Ok done

File: blocks/footer/footer.css (hunk 11 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Ok I replaced it with the corresponding variable

File: blocks/footer/footer.css (hunk 2 above)
Comment (solaris007) [reply blocks/footer/footer.css]: all good, i was not doubting the padding, but some whitespace on that line
Screenshot 2023-01-16 at 16 13 39

File: blocks/footer/footer.css (hunk 6 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Ok did add spaces

File: blocks/footer/footer.css (hunk 8 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: This is the size at which the two elements from the flex grid join and begin to be wrapped, after that size the two elements wrap and we need to change the way content is justify

File: blocks/footer/footer.css (hunk 9 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: As the are not using flex to handle the spacing between the fist link and the second sort of navigation bar, this was the way I found to handle better the same behaviour that is shown on the original footer.

File: blocks/footer/footer.css (hunk 10 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: ok changed this to a variable

File: blocks/footer/footer.css (hunk 11 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Ok changed this to the base styles

File: blocks/footer/footer.css (hunk 12 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Ok changed this to the base styles

File: blocks/footer/footer.css (hunk 13 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Ok changed this to the base styles

File: blocks/footer/footer.css (hunk 14 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Ok changed this to vairable

File: blocks/footer/footer.css (hunk 15 above)
Comment (cuzuco2) [reply blocks/footer/footer.css]: Ok changed this to the base ones.

File: blocks/footer/footer.js (hunk 7 above)
Comment (cuzuco2) [reply blocks/footer/footer.js]: Ok, did put the code in a function outside

**Comment by cuzuco2** [comment]
@solaris007 fixed the code according to your review, there is just this padding question and the wrapping size that are still open, let me know what you think of it

File: blocks/footer/footer.css (hunk 8 above)
Comment (solaris007) [reply blocks/footer/footer.css]: i see several issues:

not centered:
Screenshot 2023-01-16 at 16 29 21

works fine without padding left:
Screenshot 2023-01-16 at 16 37 51

additional flex props:
Screenshot 2023-01-16 at 16 38 18

without media query:
Screenshot 2023-01-16 at 16 38 39

lets keep it simple?

File: blocks/footer/footer.css (hunk 9 above)
Comment (solaris007) [reply blocks/footer/footer.css]: please revisit, maybe in conjunction with this comment: https://github.com/hlxsites/elco-aerotop-sg/pull/26#discussion_r1071375293

**Review by solaris007 — APPROVED** [review]

**Pull request merged**